framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -192,6 +210,8 @@ protected boolean isSideEffectFree(
      *   <li>If the method is side-effect-free (as indicated by {@link
      *       org.checkerframework.dataflow.qual.SideEffectFree} or {@link
      *       org.checkerframework.dataflow.qual.Pure}), then no information needs to be removed.
+     *   <li>If the method side effects any expression (specified as annotation values of
```

Comment:
"side effects any expression" sounds like the following case.  I think you mean, "if the method side-effects only a limited set" or "at most a limited set" of expressions.